Core/src/org/sleuthkit/autopsy/modules/embeddedfileextractor/SevenZipExtractor.java

```diff
@@ -244,216 +247,380 @@ private ArchiveFormat get7ZipOptions(AbstractFile archiveFile) {
-        int processedItems = 0;
-
-        boolean progressStarted = false;
         try {
             stream = new SevenZipContentReadStream(new ReadContentInputStream(archiveFile));
-
             // for RAR files we need to open them explicitly as RAR. Otherwise, if there is a ZIP archive inside RAR archive
             // it will be opened incorrectly when using 7zip's built-in auto-detect functionality.
             // All other archive formats are still opened using 7zip built-in auto-detect functionality.
             ArchiveFormat options = get7ZipOptions(archiveFile);
-            inArchive = SevenZip.openInArchive(options, stream);
-
+            if (password != null) {
+                inArchive = SevenZip.openInArchive(options, stream);
+            } else {
+                inArchive = SevenZip.openInArchive(options, stream, password);
+            }
             int numItems = inArchive.getNumberOfItems();
-            logger.log(Level.INFO, "Count of items in archive: {0}: {1}", new Object[]{archiveFilePath, numItems}); //NON-NLS
+            logger.log(Level.INFO, "Count of items in archive: {0}: {1}", new Object[]{escapedArchiveFilePath, numItems}); //NON-NLS
             progress.start(numItems);
             progressStarted = true;
-
             final ISimpleInArchive simpleInArchive = inArchive.getSimpleInterface();
 
             //setup the archive local root folder
             final String uniqueArchiveFileName = FileUtil.escapeFileName(EmbeddedFileExtractorIngestModule.getUniqueName(archiveFile));
-            final String localRootAbsPath = getLocalRootAbsPath(uniqueArchiveFileName);
-            final File localRoot = new File(localRootAbsPath);
-            if (!localRoot.exists()) {
-                try {
-                    localRoot.mkdirs();
-                } catch (SecurityException e) {
-                    logger.log(Level.SEVERE, "Error setting up output path for archive root: {0}", localRootAbsPath); //NON-NLS
-                    //bail
-                    return;
-                }
+            try {
+                makeLocalDirectories(uniqueArchiveFileName);
+            } catch (SecurityException e) {
+                logger.log(Level.SEVERE, "Error setting up output path for archive root: {0}", getLocalRootAbsPath(uniqueArchiveFileName)); //NON-NLS
+                //bail
+                unpackSuccessful = false;
+                return unpackSuccessful;
             }
 
             //initialize tree hierarchy to keep track of unpacked file structure
             SevenZipExtractor.UnpackedTree unpackedTree = new SevenZipExtractor.UnpackedTree(moduleDirRelative + "/" + uniqueArchiveFileName, archiveFile);
 
-            long freeDiskSpace = services.getFreeDiskSpace();
-
+            //WJS-TODO figure out size
+            long freeDiskSpace = 80000000000000L;// services.getFreeDiskSpace();
```

Comment:
Can you add a comment about why you changed from getFreeDiskSpace to the hard coded value?

Reply:
Ooops I forgot to go back and fix that, will straighten out that and the other issues.  Thank you.